### PR TITLE
fix sidebar scrolling for small displays

### PR DESCRIPTION
**Pull request by szaimen**
Fix https://github.com/nextcloud/server/issues/34777

| Before | After |
|---|---|
| image | image |

Signed-off-by: Simon L <szaimen@e.mail.de>

### CODE REVIEW

**Comment by raimund-schluessler** [comment]
I think this is there intentionally, to keep the header visible at all times. If apps don't work well with it, it can be overwritten by the app itself.

**Comment by szaimen** [comment]
> I think this is there intentionally, to keep the header visible at all times. If apps don't work well with it, it can be overwritten by the app itself.

I have yet to be found any app that really works well with this. So I'd rather fix this in general but would be fine for me ot overwrite this in server. WDYT @nextcloud/server ?

File: src/components/NcAppSidebar/NcAppSidebarTabs.vue [hunk 1]
```diff
@@ -282,7 +282,6 @@ export default {
 .app-sidebar-tabs {
 	display: flex;
 	flex-direction: column;
-	min-height: 0;
```
[the remark this answers is not in the document]
Comment (szaimen) [reply src/components/NcAppSidebar/NcAppSidebarTabs.vue:285]: > I don't see this change being relevant to the issue though :)

You are incorrect. It indeed fixes the mentioned issue :)

**Review by Pytal — APPROVED** [review]

**Comment by danxuliu** [comment]
Making the whole sidebar scrollable is problematic when the sidebar is very tall. For example, in the Comments section of Files if there are a lot of comments, or in Talk during calls, when the chat is shown in the sidebar; in the case of Talk it is specially bad, as the newest messages are at the bottom. Due to this you need to scroll all the way down to see the messages, and you need to scroll all the way up to change to another tab (and besides being uncomfortable to use it also breaks scrolling because the older messages are loaded when you scroll to the top).

But even ignoring Talk the (in my opinion) usability concerns regarding excesive scrolling apply too to Files, for example.

Therefore I would suggest a different fix. Instead of removing `min-height: 0` it could be replaced by `height: 100%`. This way the tabs would get the height of the sidebar, so it would be scrollable if the header and the tab do not fit in the screen, but the general scrolling would end when the tab headers are at the top, at which point the internal scrolling of the tab contents would start. With this you will always be able to switch between tabs, and if you need to show the sidebar header again it would be just scrolling its own height rather than the full sidebar height.

With that fix Talk would still need some further work, because the "New message" area would be hidden and the user would need to scroll down until the tab is fully shown to see it. But even if we just end setting `min-height: 0` for Talk for the time being at least it should fix the issues in Files (I have not checked other apps, though).

However... please take all that with a pinch of salt, because my CSS is quite rusty and maybe that introduces more issues than it fixes ;-)

**Comment by nickvergessen** [comment]
As per above this is a breaking change?
So need to bump the major version?
Or can we somehow limit this or make it optional so it does not change the scrolling behaviour completely?!

**Pull request merged**